src/complex/DataStructure/DataPath.cpp

```diff
@@ -106,6 +106,28 @@ DataPath DataPath::replace(std::string_view symbol, std::string_view targetName)
   return DataPath(std::move(newPath));
 }
 
+bool DataPath::attemptRename(const DataPath& oldPath, const DataPath& newPath)
```

Comment:
I'm a little confused on this function. Why does it need to take in `oldPath`?

Reply:
This method checks if an attempt to rename is valid before applying it.

Let's say you have DataPath _Foo / Bar / Bazz_. _Foo / Bar_ was renamed to _Foo / Crobar_. _Foo / Bar_ is the old path used to check if the renaming is applicable when renaming. Meanwhile, if _Foo / Random_ was renamed, _Random_ does not match the value in the path and the rename would not be relevant to the current DataPath.